storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -731,29 +731,23 @@ bool Redo_Log_Writer::close_logfile() {
   return (true);
 }
 
-/* set encryption info of redo log on io request */
-static void set_encryption_info(IORequest &req_type) {
-  auto& encryption_metadata = log_sys->m_encryption_metadata;
-  ut_ad(encryption_metadata.m_type != Encryption::NONE);
-  req_type.encryption_algorithm(encryption_metadata.m_type);
-  req_type.encryption_key(encryption_metadata.m_key,
-                          encryption_metadata.m_key_len,
-                          encryption_metadata.m_iv);
-}
 
 bool Redo_Log_Writer::write_buffer(byte *buf, size_t len) {
   byte *write_buf = buf;
 
   if (srv_redo_log_encrypt) {
-    IORequest req_type(IORequestLogWrite);
-    set_encryption_info(req_type);
+    IORequest req_type(IORequest::WRITE);
+    req_type.get_encryption_info().set(log_sys->m_encryption_metadata);
+    ut_ad(req_type.is_encrypted());
+
     Encryption encryption(req_type.encryption_algorithm());
-    ulint dst_len = len;
-    write_buf =
-        encryption.encrypt_log(req_type, buf, len, scratch_buf, &dst_len);
-    ut_a(len == dst_len);
-  }
 
+    if (!encryption.encrypt_log(buf, len, scratch_buf)) {
+      xb::error() << "Failed to encrypt redo log";
+      return false;
+    }
+    write_buf = scratch_buf;
+  }
   if (ds_write(log_file, write_buf, len)) {
```

Comment:
Do we need write_buf at all here?
scratch_buf seems to have what we need.

Reply:
if redo log are not encrypted we would just write write_buf.